x-pack/plugins/cloud_security_posture/public/pages/configurations/findings_flyout/overview_tab.tsx

```diff
@@ -63,43 +71,54 @@ const getDetailsList = (data: CspFinding, ruleFlyoutLink: string, discoverIndexL
     title: i18n.translate('xpack.csp.findings.findingsFlyout.overviewTab.ruleTagsTitle', {
       defaultMessage: 'Rule Tags',
     }),
-    description: (
+    description: data.rule?.tags?.length ? (
```

Comment:
as we have this logic of `if smth show component otherwise "-"` should we abstract it in some generic wrapper that does exactly that? smth like `x-pack/plugins/threat_intelligence/public/modules/indicators/components/common/field_value.tsx`

Reply:
I thought about it, and didn't really feel 100% comfortable with the current outcome, the reason I did not went with a genetic component here is that it felt like some fields have different requirement than just "does this field exist". some also need another fields to exist, or have links, some have multiple display value like either a link, or plain text, or just `-`.

When I examined all the possible incomes and outcomes of this potential component, i wasn't sure if I can write something that wouldn't just boil down to passing the entire `if-then` logic. 

Taking a look at `field_value.tsx` it looks like they either show a date, or the field value as is, or `-`. this is easier when the field is displayed as a its plain value, from a brief check it looks like its mostly used in tables, which mostly makes sense. But here since its for a flyout, which some fields are rendered as custom components rather than just their value, it makes things less consistent.

We can still have this component, we can have it working as `value or -` for simple text, and pass more complicated conditions + render values for other cases. I just wasn't sure if its an optimal decision.

I would love to hear your input after I provided this extra context

Reply:
I agree that our logic might be more complex, but so far I don't see many examples where the pattern defers from rendering a text or a component or `-` based on a condition. And rendering text or components isn't very different from rendering just text. Do you have examples where logic is very much different? The only case I see is for Rule Name rendering where we render either a link or a text or "-". But I think this can still be boiled down to "render a component or '-'", it's just the component would hold the logic of rendering a link if it's available or just a text

Reply:
Yes it doable, I have another problem which is with type deferring, since the check will be handled in the FieldValue component, we will have to still use the value as if its still optional when rendering it. Some examples:

```typescript
<FieldValue 
   condition={finding.data?.rule?.description} // note that this type check will not be inferred in the next line 
   children={finding.data?.rule?.description} // still need to check if it exists here
/>```

and other components will be

```typescript
<FieldValue 
   condition={finding.data?.tags?.length} // note that this type check will not be inferred in the next line 
   children={<SomeTagsComponent tags={findings.data?.tags} />} // still need to check if it exists here
/>```

Reply:
I think we can abstact it a bit differently, so that the use is smth like this
```
const NullableRender = ({ children, fallback = '-' }) => {
  if (!children) {
    return <>{fallback}</>;
  }
  return <>{children}</>;
};

<NullableRender>
        {data.rule?.tags?.length ? (
          <>
            {data.rule.tags.map((tag) => (
              <EuiBadge key={tag}>{tag}</EuiBadge>
            ))}
          </>
        ) : null}
</NullableRender>
```
it's not doing match rather than abstracting the `-` part, but at least if we decide to change the rendering logic, we wouldn't need to go through all the code in the search for dashes. It also supports just a sting as a child

Reply:
At this point if all it does is make sure use the same value, wouldn't it be better to just set `-` as a const? 
`const EmptyValue = "-"` and just use it instead of a custom component which requires to be familiar with its behaviour? it would achieve the same goal to maintain all values from a single place while also being explicit

Reply:
I don't have a strong argument for one approach over another from the top of my head. For me, it feels more natural and more React way which is ofc subjective. Having EmptyValue sprinkled all over the codebase isn't very neat. A separate component allows for the composition better in my opinion

For example, with
```
rule?.default_value ? (
      <CspFlyoutMarkdown>{rule.default_value}</CspFlyoutMarkdown>
    ) : EmptyValue
```
You can add logic to return `null` to `CspFlyoutMarkdown` itself, so it becomes a safe component by default without adding the logic of returning `EmptyValue`. Smth like this
```
export const CspFlyoutMarkdown: React.FC<PropsOf<typeof EuiMarkdownFormat>> = (props) => (
  return props.children ? <EuiMarkdownFormat textSize="s" {...props} /> : null
);

<NullableRender>
  <CspFlyoutMarkdown>{rule.default_value}</CspFlyoutMarkdown>
</NullableRender>
```
and then `CspFlyoutMarkdown` can be used in other contexts, where we don't need to render anything if it's empty.

But as I said, I don't see it as a blocker

Reply:
Thanks for sharing your perspective, I do learn a lot from it :) 
I think i will go with the `EmtpyValue` solution in the end, reason being is that it both accomplishes the reasonable need to be able to control all values form a single source and is explicit and straight forward. Personally, `NullableRender` seems to me like a bit of an overkill and still requires all children to return a nullable value only then to be changed to another value. i think we can skip the middle step and go straight to the value it self.

Will push the change now